### PR TITLE
FE MP cache and table column extension 

### PR DESCRIPTION
**Pull request by supproduction**
## PR Description

TBD

## Checklist
- [ ] I agree with the Code Contribution License Agreement in CONTRIBUTING.md

### CODE REVIEW

File: docs/marketplace/dev/frontend/table-column-type-extension.md [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+  /** Value must be an array of other types. May be recursive */
+  AnyOf = 'anyOf',
+}
+```
+
+## Overview
+
+Column Type is essentially an Angular Component that encapsulates how specific type of the column is going to be rendered withing a table column.
+
+You can configure any action by table config.
```
Comment (gund) [inline docs/marketplace/dev/frontend/table-column-type-extension.md:45]:
```suggestion
You can configure columns in table config.
```

File: docs/marketplace/dev/frontend/table-column-type-extension.md [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+</spy-table>
+```
+
+## Main Services/Decorators/Components
+
+The table module provides an opportunity to register any table column by key via static method `TableModule.withColumnComponents()`. It assigns the object of columns to the `TableColumnComponentsToken` under the hood.
+
+### ColumnTypeOption decorator
+
+Allows to validate property in runtime. All properties is showed in the `ColumnTypeOptions` interface
```
Comment (gund) [inline docs/marketplace/dev/frontend/table-column-type-extension.md:73]:
```suggestion
Allows to validate column configuration properties in runtime. All properties is showed in the `ColumnTypeOptions` interface
```

File: docs/marketplace/dev/frontend/table-column-type-extension.md [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+
+### TableColumnTypeComponent decorator
+
+Merges all default config parameters from argument with dynamic config parameters which come from table config
+
+## TableColumnRendererComponent
+
+Allows to insert data/config and context to the specific column type while config/data/context have been changed under the hood.
```
Comment (gund) [inline docs/marketplace/dev/frontend/table-column-type-extension.md:81]:
```suggestion
Component that is used by table to render every column using configuration and data.
```

File: docs/marketplace/dev/frontend/table-column-type-extension.md [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+
+## Table Column
+
+Table Column is an Angular Component that must implement specific interface (`TableColumnTypeComponent`) and be registered on the Table Module via `TableModule.withColumnComponents()` static method by giving it a string that will be associated with it when rendering table.
+
+Also you have to create own filter module and add it to the RootModule
```
Comment (gund) [inline docs/marketplace/dev/frontend/table-column-type-extension.md:87]: This seems out of place
```suggestion
```

File: docs/marketplace/dev/frontend/table-column-type-extension.md [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+
+- list - renders list of column types
+- autocomplete - renders `@spryker/input` and `@spryker/autocomplete` component
+- chip - renders `@spryker/chip` component
+- date - renders formatted date by `config`
+- dynamic - is a higher-order column that gets `ColumnConfig` from the configured `Datasource` and renders a column with retrieved `ColumnConfig`
+- image - renders image
+- input - renders `@spryker/input`component
+- select - renders `@spryker/select`component
+- text - renders static text
```
Comment (gund) [inline docs/marketplace/dev/frontend/table-column-type-extension.md:156]: We need to add column configuration with description for each table column.

**Review by gund — CHANGES_REQUESTED** [review]

File: docs/marketplace/dev/frontend/table-column-type-extension.md (hunk 5 above)
Comment (supproduction) [reply docs/marketplace/dev/frontend/table-column-type-extension.md:156]: I think it should be separates docs, which can be covered by other devs.

Comment (gund) [reply docs/marketplace/dev/frontend/table-column-type-extension.md:156]: Agreed, then we can merge as is.

**Review by gund — APPROVED** [review]

**Pull request merged**